docs/truefoundryRagArchitecture.md

```diff
@@ -0,0 +1,126 @@
+# High Level RAG Architecture
```

Comment:
Name of md files should ideally be all caps, here it can be just `ARCHITECTURE.md`